python/pyspark/sql/datasource.py

```diff
@@ -16,6 +16,7 @@
 #
 from abc import ABC, abstractmethod
 from collections import UserDict
+import pyarrow as pa
```

Comment:
Hmm I don't think we can directly import pyarrow here unless we make pyarrow the one of the default dependency for pyspark. cc @HyukjinKwon WDYT?

Reply:
yeah let's don't import here

Reply:
You can do with `if typing.TYPE_CHECKING` and import pyarrow as pa, then type it as `"pa.RecordBatch"` below.